monai/transforms/utils.py

```diff
@@ -1171,6 +1176,247 @@ def get_largest_connected_component_mask(
+                )
+                f, b = max(p[2].round().int().item() - margin, 0), min(
+                    p[2].round().int().item() + margin + 1, features.shape[-1]
+                )
+                if (features[bs, 0, left:right, t:d, f:b] > 0).any():
+                    index = features[bs, 0, left:right, t:d, f:b].max()
+                    outs[[bs]] += lib.isin(features[[bs]], index)
+                    break
+    outs[outs > 1] = 1
+    return convert_to_dst_type(outs, dst=img_pos, dtype=outs.dtype)[0]
+
+def convert_points_to_disc(image_size, point, point_label, radius=2, disc=False):
+    """
+    Convert a 3D point coordinates into image mask. The returned mask has the same spatial
+    size as `image_size` while the batch dimension is the same as point' batch dimension.
+    The point is converted to a mask ball with radius defined by `radius`.
+    Args:
+        image_size: The output size of th
+        point: [b, N, 3]
+        point_label: [b, N]
+        radius: disc ball radius size
+        disc: If true, use regular disc other other use gaussian.
+    """
+    if not torch.is_tensor(point):
+        point = torch.from_numpy(point)
+    masks = torch.zeros(
+        [point.shape[0], 2, image_size[0], image_size[1], image_size[2]],
+        device=point.device,
+    )
+    row_array = torch.arange(
+        start=0, end=image_size[0], step=1, dtype=torch.float32, device=point.device
+    )
+    col_array = torch.arange(
+        start=0, end=image_size[1], step=1, dtype=torch.float32, device=point.device
+    )
+    z_array = torch.arange(
+        start=0, end=image_size[2], step=1, dtype=torch.float32, device=point.device
+    )
+    coord_rows, coord_cols, coord_z = torch.meshgrid(z_array, col_array, row_array)
+    # [1,3,h,w,d] -> [b, 2, 3, h,w,d]
+    coords = (
+        torch.stack((coord_rows, coord_cols, coord_z), dim=0)
+        .unsqueeze(0)
+        .unsqueeze(0)
+        .repeat(point.shape[0], 2, 1, 1, 1, 1)
+    )
+    for b in range(point.shape[0]):
+        for n in range(point.shape[1]):
+            if point_label[b, n] > -1:
+                channel = 0 if (point_label[b, n] == 0 or point_label[b, n] == 2) else 1
+                if disc:
+                    masks[b, channel] += (
+                        torch.pow(
+                            coords[b, channel]
+                            - point[b, n].unsqueeze(-1).unsqueeze(-1).unsqueeze(-1),
+                            2,
+                        ).sum(0)
+                        < radius**2
+                    )
+                else:
+                    masks[b, channel] += torch.exp(
+                        -torch.pow(
+                            coords[b, channel]
+                            - point[b, n].unsqueeze(-1).unsqueeze(-1).unsqueeze(-1),
+                            2,
+                        ).sum(0)
+                        / (2 * radius**2)
+                    )
```

Comment:
```suggestion
            point_bn = unsqueeze_right(point[b, n], 3)
            pow_diff = torch.pow( coords[b, channel] - point_bn, 2).sum(0)
            if point_label[b, n] > -1:
                channel = 0 if (point_label[b, n] == 0 or point_label[b, n] == 2) else 1
                if disc:
                    masks[b, channel] += pow_diff < radius**2
                else:
                    masks[b, channel] += torch.exp(-pow_diff / (2 * radius**2))
```
Something like this is more readable?